packages/wallet/src/redux/protocols/concluding/states.ts

```diff
@@ -1,55 +1,63 @@
 import { InstigatorConcludingState, isConcludingInstigatorState } from './instigator/states';
 import { ResponderConcludingState, isConcludingResponderState } from './responder/states';
-import { Constructor } from '../../utils';
+import { StateConstructor } from '../../utils';
 import { ProtocolState } from '..';
 
 export * from './instigator/states';
 export * from './responder/states';
 
-export type ConcludingState = InstigatorConcludingState | ResponderConcludingState;
-
-export type TerminalState = Success | Failure;
-
-export type FailureReason =
-  | 'NotYourTurn'
-  | 'ChannelDoesntExist'
-  | 'ConcludeCancelled'
-  | 'DefundFailed';
+// -------
+// States
+// -------
 
 export interface Failure {
-  type: 'Failure';
+  type: 'Concluding.Failure';
```

Comment:
I think that containers still need information about the states at runtime: otherwise they can be confused by, e.g. `Concluding.Acknowledge` and `Challenging.Acknowledge` ? The protocol container, for example, does consume `ProtocolState` -- a union:

```typescript
export type ProtocolState =
  | ApplicationState
  | IndirectFundingState
  | DirectFundingState
  | WithdrawalState
  | ResponderState
  | FundingState
  | DefundingState
  | ChallengerState
  | ConcludingState
  | IndirectDefundingState
  | TransactionSubmissionState;
```

Reply:
Ok - that makes sense. There are probably ways to get around this but they might not be worth it. This seems like a perfectly reasonable solution for now.